Sources/Property.swift

```diff
@@ -22,6 +34,10 @@ public protocol PropertyProtocol: class, BindingSourceProtocol {
 	/// completes when the property has deinitialized, or has no further
 	/// change.
 	var signal: Signal<Value, NoError> { get }
+
+	/// Perform an arbitrary action on the current value of the property.
+	/// Serialized properties should invoke `body` in its protected section.
+	func withValue<R>(_ body: (Value) throws -> R) rethrows -> R
```

Comment:
Would it make more sense to add something similar to `startWithSignal`?

```
public func observe(_ handler: (Value, Signal<Value, NoError>) -> ())
```

Then there wouldn't be so many requirements?